main.go

```diff
@@ -97,12 +106,22 @@ func main() {
 	// returns nil if not provided and falls back to simple TCP.
 	tlsConfig := createTLSConfig(*esCA, *esClientCert, *esClientPrivateKey, *esInsecureSkipVerify)
 
+	defaultTransport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+		Proxy:           http.ProxyFromEnvironment,
+	}
+
 	httpClient := &http.Client{
-		Timeout: *esTimeout,
-		Transport: &http.Transport{
-			TLSClientConfig: tlsConfig,
-			Proxy:           http.ProxyFromEnvironment,
-		},
+		Timeout:   *esTimeout,
+		Transport: defaultTransport,
+	}
+
+	if esAWS != nil {
+		httpClient.Transport = &roundtripper.AWSSigningTransport{
+			DefaultTransport: defaultTransport,
+			Credentials:      credentials.NewChainCredentials([]credentials.Provider{&credentials.EnvProvider{}, &credentials.SharedCredentialsProvider{}}),
```

Comment:
These changes are needed for IRSA (IAM roles for service accounts) to work
```suggestion
			Credentials: credentials.NewChainCredentials([]credentials.Provider{
				&credentials.EnvProvider{},
				&credentials.SharedCredentialsProvider{},
				stscreds.NewWebIdentityRoleProvider(
					sts.New(sess),
					os.Getenv("AWS_ROLE_ARN"),
					"",
					os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE"),
				),
				&ec2rolecreds.EC2RoleProvider{
					Client: ec2metadata.New(sess),
				},
			}),
```

Reply:
@JackFazackerley could you accept these changes so we can at least get this PR in a working state? Thanks!